packages/client/src/exampleTypes.js

```diff
@@ -126,6 +126,14 @@ export default F.stampKey('type', {
     },
   },
   results: {
+    init(node, extend, snapshot) {
+      if (_.isNil(node.hasResults))
+        Object.defineProperty(node, 'hasResults', {
```

Comment:
Same as below - I think this makes more sense as a new core feature that types can opt into